### PR TITLE
Molecule

### PR DESCRIPTION
**Pull request by rpelisse**

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -24,6 +31,39 @@ add_parent_volume_if_provided() {
...
+   if [ -n "${TOOLS_MOUNT}" ]; then
+     echo "-v ${TOOLS_DIR}:${TOOLS_MOUNT}:ro"
+   fi
+ else
+   echo "Warning: Provided tools dir ${TOOLS_DIR} does not exist, won't be added to container's volume." >&2
+ fi
+}
+
+container_user_if_enabled() {
+  if [ -n "${CONTAINER_USERNAME}" ]; then
```
Comment (spyrkob) [inline run.sh:57]: why is `CONTAINER_USERNAME` needed? it doesn't seem to be used anywhere

File: run.sh [hunk 2]
```diff
@@ -41,14 +81,12 @@ readonly CONTAINER_COMMAND=${CONTAINER_COMMAND:-"${WORKSPACE}/hera/wait.sh"}
...
-            --workdir ${WORKSPACE} \
-            -v "${JOB_DIR}":${WORKSPACE}:rw \
-            -v /opt/:/opt/:ro \
-            -v "${JENKINS_ACCOUNT_DIR}/.ssh/":/var/jenkins_home/.ssh/:ro \
+            --name "${CONTAINER_TO_RUN_NAME}" $(container_user_if_enabled) \
+            --add-host=${CONTAINER_SERVER_HOSTNAME}:${CONTAINER_SERVER_IP}  \
+            --rm $(add_parent_volume_if_provided) $(systemd_if_enabled) \
+            --workdir ${WORKSPACE} $(add_ports_if_provided) \
+            -v "${JOB_DIR}":${WORKSPACE}:rw $(mount_tools_if_provided)\
+            -v "${JENKINS_ACCOUNT_DIR}/.ssh/":/var/jenkins_home/.ssh/:rw \
```
Comment (spyrkob) [inline run.sh:89]: why the change to `rw`?

File: inspect.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+readonly BUILD_SCRIPT=${BUILD_SCRIPT:-'1'}
+shift
```
Comment (spyrkob) [inline inspect.sh:3]: is the `shift` needed? this doesn't seem to use the parameters

File: run.sh [hunk 4]
```diff
@@ -4,14 +4,21 @@ readonly PARENT_JOB_NAME=${PARENT_JOB_NAME}
 readonly PARENT_JOB_BUILD_ID=${PARENT_JOB_BUILD_ID}
 readonly BUILD_PODMAN_IMAGE=${BUILD_PODMAN_IMAGE:-'localhost/automatons'}
 readonly JENKINS_HOME_DIR=${JENKINS_HOME_DIR:-'/home/jenkins/'}
+readonly JENKINS_CONTAINER_HOME_DIR=${JENKINS_CONTAINER_HOME_DIR:-'/var/jenkins_home/'}
 readonly JENKINS_ACCOUNT_DIR=${JENKINS_ACCOUNT_DIR:-'/home/jenkins'}
-readonly JENKINS_UID=${JENKINS_UID:-'1000'}
-readonly JENKINS_GUID=${JENKINS_GUID:-"${JENKINS_UID}"}
+readonly CONTAINER_USERNAME=${CONTAINER_USERNAME:-'jenkins'}
```
Comment (rpelisse) [inline run.sh:9]: @spyrkob CONTAINER_USERNAME is set here.
